src/cpp/shared/game/save.cpp

```diff
@@ -42,6 +42,18 @@ extern void __fastcall GenerateStandardFileName(char*, char*);
 static const char *tmpFileName = "tmp";
 static const LPCWSTR tmpFileNameW = L"tmp";
 
+template<typename T>
+void ReadArrayFromXML(T &dest, xsd::cxx::tree::sequence<ironfist_save::arrayInt_t> &src) {
+  for (int i = 0; i < src.size(); i++)
+    dest[i] = src.at(i).value();
+}
+
+template<typename T>
+void WriteArrayToXML(xsd::cxx::tree::sequence<ironfist_save::arrayInt_t> &dest, T &src) {
+  for (int i = 0; i < ELEMENTS_IN(src); i++)
```

Comment:
The behavior I expect to see is that ELEMENTS_IN(src) will always evaluate to 1.

Can you explain why this won't be the case?

Even if it works, you still have to explain it.